runtime/gc_vlhgc/HeapRegionManagerVLHGC.cpp

```diff
@@ -211,6 +211,7 @@ MM_HeapRegionManagerVLHGC::getHeapMemorySnapshot(MM_GCExtensionsBase *extensions
 		snapshot->_totalRegionEdenSize = allocateEdenTotal;
 	}
 
+	Assert_MM_true(snapshot->_totalRegionReservedSize >= (snapshot->_totalRegionEdenSize - allocateEdenTotal));
```

Comment:
Could we put a similar assert somewhere at GC end path, instead? Just before this is triggered (hence after all resing logic is complete). This will be tested only if MXBean is activated, but should always be.

Reply:
Could we do it even later, very close to when we trigger GC-end, so it covers other invocations of calculateEdenSize, too?